encoders/device.go

```diff
@@ -0,0 +1,34 @@
+package encoders
+
+type Device struct {
```

Comment:
Why do we abstract away the underlying encoder into a Device type? Seems like a big ask to say "all future encoders will behave this way". In fact I'm not sure I'd be able to implement this interface with the encoders I've worked with (half arc+small slits encoder)

Reply:
A suggestion: Say we implement `QuadratureRotary` type- it could implement the methods `Value` and `SetValue`, which then could be used by users who define the `Encoder` interface:

```go
type Encoder interface {
     Value() int32
     SetValue(int32) error
}
```
This way the code is simpler and users are able to choose their level of abstraction. As a forethought, I would not rush into defining the interface on our side just yet- let users experiment and see what works best!

As a final comment I also suggest using more evocative method names: `Ticks`/`SetTicks`? Or `Position`/`SetPosition`? `Increments`/`SetIncrements`? I feel these all convey more domain specific ideas than `Value`.

Reply:
Ok good point ... I agree with what you're saying, I was just thinking of the `Device` as rotary device.  The abstraction was meant to allow for implementing different methods for counting and debouncing the position.

Reply:
@soypat  I've updated the naming according to your suggestions